### PR TITLE
Release/1.12.0

### PR DESCRIPTION
**Pull request by mchowning**
Merging release/1.12.0 branch into master

#### Remaining work
- [X] Merge gutenberg side (there is nothing to merge because `rnmobile/release-1.12.0` hasn't been updated since it was branched off of `rnmobile/master` ([branch comparison](https://github.com/WordPress/gutenberg/compare/rnmobile/master...rnmobile/release-1.12.0)).
- [x] Test and approve [WPiOS PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/12491)
- [x] Test and approve [WPAndroid PR](https://github.com/wordpress-mobile/WordPress-Android/pull/10490)
- [ ] Merge this PR
- [ ] Tag Gutenberg
- [ ] Tag head of master branch for 1.12.0 release
- [ ] Update iOS PR to reference gutenberg-mobile release tag and merge iOS PR
- [ ] Update Android PR to reference tagged commit and merge Android PR
- [ ] Open PR from master to develop (won't be necessary unless we end up adding changes to this PR)
- [ ]  🎉

### Please let me know if you think I am missing any steps ⤴️

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,3 +1,10 @@
+1.12.0
+------
+* Add rich text styling to video captions
+* Prevent keyboard dismissal when switching between caption and text block on Android
+* Blocks that would be replaced are now hidden when add block bottom sheet displays
+* Tapping on empty editor area now always inserts new block at end of post
+
```
Comment (marecar3) [inline RELEASE-NOTES.txt]: Not sure if we are missing Media&Text and Group blocks from this list, or we want to hide them in 1.12.0 release?

Comment (mchowning) [reply RELEASE-NOTES.txt]: I confirmed with @etoledom that we do not want either of those blocks to be available in this release, [and they are guarded by a dev flag](https://github.com/WordPress/gutenberg/blob/rnmobile%2Frelease-1.12.0/packages%2Fblock-library%2Fsrc%2Findex.native.js#L146). 

I also made release builds of both iOS and Android and confirmed that those blocks are not selectable from the bottom modal block picker and if a post is shown on mobile that contains those blocks (created on the web), then the app just displays a block placeholder. Sounds like you were observing something different though, would you like to double-check this before we merge @marecar3 ?

Comment (marecar3) [reply RELEASE-NOTES.txt]: Thanks @mchowning for confirmation.
It's always good to check it :)

Comment (mchowning) [reply RELEASE-NOTES.txt]: >It's always good to check it :)

Absolutely! Thanks for bringing it up!

**Pull request merged**